uilib/burnsimManager.py

```diff
@@ -0,0 +1,243 @@
+
+                    if child.attrib['EndsInhibited'] == '1':
+                        motor.grains[-1].setProperty('inhibitedEnds', 'Top')
+                    elif child.attrib['EndsInhibited'] == '2':
+                        motor.grains[-1].setProperty('inhibitedEnds', 'Both')
+
+                    if grainType in ('1', '3', '7'): # Grains with core diameter
+                        motor.grains[-1].setProperty('coreDiameter', inToM(child.attrib['CoreDiameter']))
+
+                    if grainType == '2': # D grain specific properties
+                        motor.grains[-1].setProperty('slotOffset', inToM(child.attrib['EdgeOffset']))
+
+                    elif grainType == '3': # Moonburner specific properties
+                        motor.grains[-1].setProperty('coreOffset', inToM(child.attrib['CoreOffset']))
+
+                    elif grainType == '5': # C grain specific properties
+                        motor.grains[-1].setProperty('slotWidth', inToM(child.attrib['SlotWidth']))
+                        radius = motor.grains[-1].getProperty('diameter') / 2
+                        motor.grains[-1].setProperty('slotOffset', radius - inToM(child.attrib['SlotDepth']))
+
+                    elif grainType == '6': # X core specific properties
+                        motor.grains[-1].setProperty('slotWidth', inToM(child.attrib['SlotWidth']))
+                        motor.grains[-1].setProperty('slotLength', inToM(child.attrib['CoreDiameter']) / 2)
+
+                    elif grainType == '7': # Finocyl specific properties
+                        motor.grains[-1].setProperty('finWidth', inToM(child.attrib['FinWidth']))
+                        motor.grains[-1].setProperty('finLength', inToM(child.attrib['FinLength']))
+                        motor.grains[-1].setProperty('numFins', int(child.attrib['FinCount']))
+
+                    if not propSet: # Use propellant numbers from the forward grain
+                        impProp = child.find('Propellant')
+                        propellant = motorlib.propellant()
+                        propellant.setProperty('name', impProp.attrib['Name'])
+                        n = float(impProp.attrib['BallisticN'])
+                        a = float(impProp.attrib['BallisticA']) * 1/(6895**n)
+                        propellant.setProperty('n', n)
+                        propellant.setProperty('a', motorlib.convert(a, 'in/(s*psi^n)', 'm/(s*Pa^n)')) # Conversion only does in/s to m/s, the rest is handled above
+                        propellant.setProperty('density', motorlib.convert(float(impProp.attrib['Density']), 'lb/in^3', 'kg/m^3'))
+                        propellant.setProperty('k', float(impProp.attrib['SpecificHeatRatio']))
+                        propellant.setProperty('m', 23.67) # BurnSim does't provide this value
+                        propellant.setProperty('t', 3500) # Or this one. TODO: modulate this to make the ISP correct
```

Comment:
although BurnSim's combustion temp is usually zero, it does provide molar mass if the user entered it, so we should import it:
```xml
<Propellant Name="68/10" Density="0.058" ISPStar="146.3" BallisticA="0.0341" BallisticN="0.2249" SpecificHeatRatio="1.226" MolarMass="21.57" CombustionTemp="0">
```

We can assume `m=23.67` (where did you get that value, btw?) and increase/decrease combustion temp based on `IspStar` (as you've done in the exporter), or we could just refactor and use C* directly.

 I was going to make a separate issue about this but most propellant numbers I've seen list only C* or Isp*, so we should probably just use that number internally. We end up calculating C* from T and m anyways every single iteration, so it would be a slightly more performant to do so. And fwiw in the motor code I can't find anything else T is used for...

Reply:
The number came from ProPEP and is the molar mass for the default propellant. I changed it to use the value from the file if it isn't 0, and the default otherwise. I think you are correct about the direction we should head in for `c*` so I didn't bother executing my plan to change `t` to get `c*` to match `isp*`. I included `t` separately because some of the reading I did about erosive burning (RPE) suggested that temperature plays a role. I don't know if it is worth keeping around if it is only ever used to calculate `c*`.